### PR TITLE
upgrade: refactor upgrade tests

### PR DESCRIPTION
**Pull request by philip-stoev**
* remove all mentions of "current version" or "previous version" as they were confusing.
  Instead, think in terms of the Mz instance we are upgrading FROM and the Mz instance
  we are upgrading TO

* remove testing of upgrade from the materialized/latest container, as Brandon rightfully
  observed, it is the same as testing the latest tagged Mz version

* introduce a separate sub-workflow that tests upgrading from the current source to the
  current source, so that one can write upgrade tests about unreleased features as they
  are being developed in a branch.

* update mkrelease.py to account for the changes above.

* Update README.md accordingly

* Add v0.9.0 to the list of versions to upgrade from

------

@quodlibetor I was unable to test the `mkrelease.py` changes as this script does not run for me locally. I also had to add `v0.9.0` manually to the `mzcompose.yml` file

@benesch I understand the upgrade test has stretched the `mzcompose.yml` file to the limit ... Hopefully the next time this is touched will be in order to rewrite it in python. I did not do that now as I was not confident that I will be able to replicate all the required functionality from YAML to Python within a reasonable time.

### CODE REVIEW

File: test/upgrade/README.md [hunk 1]
```diff
@@ -45,20 +47,22 @@ There are also two other special version identifiers:
 2. Add a section:
 
 ```
-  upgrade-from-0_6_1:
```
Comment (philip-stoev) [inline test/upgrade/README.md:48]: This is in the README.md

**Comment by quodlibetor** [comment]
> I was unable to test the mkrelease.py changes as this script does not run for me locally.

That's too bad, it should, how did it error?

**Review by benesch — APPROVED** [review]

**Review by quodlibetor — APPROVED** [review]
Thanks, this makes everything much clearer.

File: test/upgrade/check-from-any_version-smoke.td [hunk 2]
```diff
@@ -1,15 +0,0 @@
-# Copyright Materialize, Inc. and contributors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (philip-stoev) [reply test/upgrade/check-from-any_version-smoke.td:1]: Yes, I used them for the initial PR, I should have removed them as soon as I had actual test content in place.

[the remark this answers is not in the document]
Comment (quodlibetor) [reply test/upgrade/check-from-any_version-smoke.td:1]: confirmed offline.

**Pull request merged**